src/components/UserSelector/UserSelector.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { User } from '../../types/User';
+
+type Props = {
+  users: User[];
+  selectedUserId: number;
+  handleSelectUser: (userId: number) => void;
+};
+export const UserSelector: React.FC<Props> = React.memo(({
+  users,
+  selectedUserId,
+  handleSelectUser,
+}) => {
+  const dropdownRef = useRef<HTMLDivElement>(null);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const selectedUser = users.find(user => user.id === selectedUserId);
+
+  const handleClick = (userId: number) => {
+    handleSelectUser(userId);
+    setIsDropdownOpen(false);
+  };
+
+  const handleClickOutside = (event: MouseEvent) => {
+    if (dropdownRef.current
+      && !dropdownRef.current.contains(event.target as Node)) {
```

Comment:
```suggestion
    if (isDropdownOpen) {
```

Reply:
in this case it doesn't work

Reply:
yeah sorry, I realized why, it wasn't that obvious